whd/dispatcher.py

```diff
@@ -395,6 +402,24 @@ def dispatch_pullrequest_event(
             return False
 
         def _process_pr_event(**kwargs):
+            if not (github_helper := self._github_api_for_pr_event(pr_event)):
+                logger.error(
+                        f'Unable to create github-api for PR #{pr_event.number()} of '
+                        f'repository {pr_event.repository().repository_path()}.'
+                    )
+                return
+
+            if (
+                self._pr_modified_pipeline_definitions(pr_event)
+                and pr_event.action() in [PullRequestAction.OPENED, PullRequestAction.SYNCHRONIZE]
+            ):
+                logger.info(f'Validating .ci/pipeline-definition for PR #{pr_event.number()}')
+                self._validate_pipeline_definitions(
```

Comment:
generally speaking, how about moving the whole code that will check a PR for pipeline-definition-errs into a separate module (e.g. whd/pr_validator.py or whd/pr.py or whd/validate.py), so dispatcher.py can focus on, well, dispatching..